core/arch/arm/kernel/boot.c

```diff
@@ -943,7 +943,7 @@ static int get_memory_all(void *fdt, struct core_mmu_phys_mem *mem)
 		if (offs < 0)
 			break;
 
-		if (_fdt_get_status(fdt, offs) <= 0)
+		if (_fdt_get_status(fdt, offs) != DT_STATUS_OK_NSEC)
```

Comment:
I think it should be `!(_fdt_get_status(fdt, offs) & DT_STATUS_OK_NSEC)`.
Would match `status = "okay"` and whatever value for `secure-status`.

Reply:
By the way, I think we should fix `_fdt_get_status()` description to what is actually does:
```patch
 /*
  * Read the status and secure-status properties into a bitfield.
-  * @status is set to DT_STATUS_DISABLED or a combination of DT_STATUS_OK_NSEC
+  * Return DT_STATUS_DISABLED or a combination of DT_STATUS_OK_NSEC
  * and DT_STATUS_OK_SEC
- * Returns positive or null status value on success or -1 in case of error.
  */
 int _fdt_get_status(const void *fdt, int offs);
```

Reply:
I agree that the test is wrong, but I don't think what you are proposing is correct either.
If `secure-status="disabled"` then OP-TEE should not access the memory, so in fact the only acceptable value is `(DT_STATUS_OK_NSEC | DT_STATUS_OK_SEC)`.
So this should be: `if (_fdt_get_status(fdt, offs) != (DT_STATUS_OK_NSEC | DT_STATUS_OK_SEC))`. What do you think?

Reply:
This function looks for memory that non-secure can access aside whereas secure world will access it or not. So I still think only bit flag `DT_STATUS_OK_NSEC` should be looked at. Does that make sense?

Reply:
I agree with you @jforissier 
``` C
if (_fdt_get_status(fdt, offs) != (DT_STATUS_OK_NSEC | DT_STATUS_OK_SEC))
```
Looks like the way to do it.

Reply:
If `secure-status` is explicitly set to something different from ok then I guess secure world should stay clear.
If nothing special is done we're defaulting to follow the non-secure status any way.

Reply:
Ok, you're both right, this function is indeed about discovering shared memory.